draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,33 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+Implementations MUST NOT support function extensions that are not registered.
+
+Note: an implementation may behave by default in a "strict" mode that conforms to this specification
+and offer some way of behaving in a "non-strict" mode that supports unregistered function extensions.
```

Comment:
```suggestion
Note: an implementation MAY offer some way of switching between a "strict" mode that conforms to this specification, which must be the default,
and  a "non-strict" mode that supports unregistered function extensions.
```

Maybe?

Reply:
That's reasonable except using "MAY" in non-normative text feels a little over the top and "which must be the default" also seems slightly too prescriptive. How about this instead?
```suggestion
Note: an implementation may offer a default "strict" mode that conforms to this specification
and a "non-strict" mode that supports unregistered function extensions.
```

Reply:
I don't want to waste any more time discussing the unspec'd behaviour of implementations, such as what that should be like and whether it should appear in normative text, so I deleted the note.